app/controllers/interactive_controller.rb

```diff
@@ -6,20 +6,19 @@ def edit
 
   def update
     set_page
-    interactive_type_name = @interactive.class.model_name.human.downcase
-    if (@interactive.update_attributes(get_interactive_params))
-      # respond success
-      flash[:notice] = "Your #{interactive_type_name} was updated."
-    else
-      flash[:warning] = "There was a problem updating your #{interactive_type_name}."
+    input_params = get_interactive_params
+    # linked_interactives param follows ISetLinkedInteractives interface format. It isn't a regular attribute.
+    # It requires special treatment and should be removed from params before .update_attributes is called.
+    if input_params.has_key? :linked_interactives
+      linked_linteractives = input_params.delete :linked_interactives
+      if linked_linteractives.present?
+        set_linked_interactives(JSON.parse linked_linteractives)
+      end
     end
+    @interactive.update_attributes! input_params
     respond_to do |format|
-      if @page
-        update_activity_changed_by(@activity)
-        format.html { redirect_to edit_activity_page_path(@activity, @page) }
-      else
-        format.html { redirect_to :back }
```

Comment:
Was this redirect_to back not needed anymore?

Reply:
I think it was a dead code, as it would be executed only if `@page` was `nil`. I think we have only one place to edit interactives and the editor is providing page ID.

Reply:
I have a hunch that there is another place where interactives are edited. But hopefully I'm wrong.